ScreenManager.c

```diff
@@ -229,6 +229,7 @@ void ScreenManager_run(ScreenManager* this, Panel** lastFocus, int* lastKey, con
    bool rescan = false;
    int sortTimeout = 0;
    int resetSortTimeout = 5;
+   int maxIterations = this->settings->maxIterations;
```

Comment:
I'm not sure if this variable really makes things easier to read.

Reply:
`maxIterations` is being used thrice in the `if` block on line 244. I think it'll become cluttered if `this->settings->maxIterations` is used in all three places.

But maybe, line 232 can be placed between the `if` block and `force_redraw = false`. What are your thoughts on this?

Reply:
First off, `maxIterations` is a bad name for how the variable currently is used. And unless the value is stored back into the setting, this doesn't help with reducing the scope.

If you, on the second note though, just use the fully qualified reference to the field in the settings, you can skip line 232 entirely and just need the three lines below. You could even put the decrement and the second check into one line:

```c
if (this->settings->maxIterations != -1) {
   if (!--this->settings->maxIterations) {
      // Exit the program
   }
}
```

Regarding the name: while the setting is called `max-iterations` the internal field could better be called `iterationsLeft` or something along those lines.

Reply:
Hi. You're right. `iterationsLeft` is a better name given the way it is being handled. I'll change this.